### PR TITLE
Add endpoint to get all comments for an article

### PR DESCRIPTION
**Pull request by jasonbarrows**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -132,3 +132,55 @@ describe('GET /api/articles/:article_id', () => {
...
+      });
+  });
+
+  it('200: responds with an empty array if the article id exists but has no comments', () => {
+    return request(app)
+      .get('/api/articles/2/comments')
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+
```
Comment (lewispricey) [inline __tests__/app.test.js:165]: This is a really good test, Might be worth adding in an extra assertion here that checks comments is an instance of an array

File: models/articles.models.js [hunk 2]
```diff
@@ -22,3 +22,17 @@ exports.selectArticleById = (article_id) => {
     return rows[0];
   });
 };
+
+exports.selectAllCommentsByArticleId = (article_id) => {
+  return Promise.all([
```
Comment (lewispricey) [inline models/articles.models.js:27]: Great use of promise.all here to differentiate between a valid empty response and an article that doesn't exist!

**Review by lewispricey — APPROVED** [review]
Awesome work on this ticket Jason. I've added a little comment about one assertion I'd add to the valid but no comments test but other than that you're good to go.

Merge away once you've added the extra check 🚀

File: __tests__/app.test.js (hunk 1 above)
Comment (jasonbarrows) [reply __tests__/app.test.js:165]: I have added the extra assertion as per your suggestion. Thanks! 😄

**Pull request merged**